nixos/modules/services/web-apps/pairdrop.nix

```diff
@@ -0,0 +1,154 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    boolToString
+    getExe
+    isAttrs
+    isBool
+    maintainers
+    mapAttrs
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    optionalAttrs
+    types
+    ;
+
+  cfg = config.services.pairdrop;
+
+  json = pkgs.formats.json { };
+in
+{
+  options.services.pairdrop = {
+    enable = mkEnableOption "pairdrop";
+
+    package = mkPackageOption pkgs "pairdrop" { };
+
+    port = mkOption {
+      type = types.port;
+      default = 3000;
+      example = 3010;
+      description = "The port to listen on.";
+    };
+
+    settings = mkOption {
+      description = ''
+        Additional configuration (environment variables) for PairDrop, see
+        <https://github.com/schlagmichdoch/PairDrop/blob/master/docs/host-your-own.md#environment-variables>
+        for supported values.
+      '';
+
+      type = types.submodule {
+        freeformType =
+          with types;
+          attrsOf (oneOf [
+            bool
+            int
+            str
+          ]);
+
+        options = {
+          RTC_CONFIG = mkOption {
```

Comment:
I do wonder if this would make more sense as a top-level option instead of one inside `settings`. Also, this is not *technically* "settings" in the RFC 42 sense IMO - maybe `environment` would work better?

Reply:
I took a look at other modules in nixpkgs and both seem to exist. I guess `environment` could be better in case upstream adds an actual config file in the future.
As for RTC_CONFIG, I agree, it makes more sense for it to be a separate thing. And I'll drop the path handling altogether, if someone wants to point it to a file they can simply use `environment.RTC_CONFIG` instead :)